packages/mdc-chips/README.md

```diff
@@ -52,13 +52,39 @@ npm install --save @material/chips
 </div>
 ```
 
+#### Leading and Trailing Icons
+
+You can optionally add a leading icon (i.e. thumbnail) and/or a trailing icon to a chip. To add an icon, add the relevant class (either `mdc-chip--with-leading-icon` or `mdc-chip--with-trailing-icon`) to the chip element, add an `i` element with your preferred icon, and give it a class of `mdc-chip__icon`. If you're adding a trailing icon, also give the icon a class of `mdc-chip__icon--trailing`.
```

Comment:
The CSS styles below seem to suggest that trailing icons can be individually hovered/focused. If so, we might want to consider using a `button` element (or barring that, an `a` element which we'd cancel click events on) for trailing icon rather than an `i` element, if clicking the trailing icon has a distinct action from clicking the rest of the chip. Otherwise you're never actually going to trigger the `:focus` style below without also going out of your way to specify `tabindex`.

On the other hand, if clicking the trailing icon does the same thing as clicking the rest of the chip, I'm not sure whether the icon itself should be individually focusable at all? We should determine which element[s] should be focusable and consider updating our DOM accordingly. Right now, nothing on the chips demo page is focusable.

Reply:
Okay, I just looked at this again after seeing Andy's comments, and realized things on the demo page _are_ focusable... but it's almost imperceptible that things are getting focused unless you know exactly where you should be looking. I think we may have a design problem...

Reply:
cc @acdvorak on this since he had some reservations about my `button` element idea in particular (mainly related to needing to override more styles, and maybe a concern about issues with padding in some browsers?)

Reply:
Good spotting, I'll ask designers about the imperceptible focus issue. 

I think it's fine to specify `tabindex` for icons, that's what we do for text field too. Using a `i` element makes the most sense semantically and specifying `tabindex` is a small price to pay.

We also need to specify `tabindex` for each chip, which leads to the question of whether we should use a `button` element for a chip. But from a design/semantics standpoint we probably want to separate the concept of a button from the concept of a chip as much as possible and not confuse clients...

Reply:
I agree, chips are not buttons.

If anything, they're closer to checkboxes or radio buttons, but we can't really use those as the base element, so `<div tabindex="0">` or `<span tabindex="0">` seem reasonable to me.

Semantically, the "X" icon is a button, but it seems weird to use a `<button>` element for that, so I would also vote to stick with `<i tabindex="0" role="button">`.

Reply:
It's true that we do have other components doing the same thing, and it's true that using the `i` element for icons is pretty pervasive nowadays, so yeah, I guess this discussion can be tabled.

The upside of a more semantically-correct element is not only getting tabindex for free but also potentially ARIA role (and technically we should probably have both?). However, it might also be the case that different roles make sense for different types of chips/icons, which would also discourage settling on a specific more semantic element. (I know Andy brought this up on the choice chips PR.)

FWIW the `i` element itself is arguably not semantic at all; it originally indicated italicized text back in the 90s, and was later discouraged in lieu of other tags like `em` that _do_ have semantic meaning.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/i):

> The `<i>` represents a range of text that is set off from the normal text for some reason, for example, technical terms, foreign language phrases, or fictional character thoughts. It is typically displayed in italic type.

> Use this element only when there is not a more appropriate semantic element.